File: php/src/Logs.php
<?php

namespace Sentroy\ClientSdk;

class Logs
{
    /** @var HttpClient */
    private $http;

    public function __construct(HttpClient $http)
    {
        $this->http = $http;
    }

    /**
     * List mail-log entries. Filter by status, domain, and ISO timestamp
     * range ("from" / "to", both inclusive). Results are paginated
     * server-side; pass "page" and "limit" to walk a large window.
     *
     * @param array $params {
     *     @type int    $page     Optional
     *     @type int    $limit    Optional
     *     @type string $status   Optional — "queued", "processing", "sent", "bounced", "failed"
     *     @type string $domainId Optional
     *     @type string $from     Optional — ISO timestamp, inclusive
     *     @type string $to       Optional — ISO timestamp, inclusive
     * }
     * @return array
     */
    public function getAll(array $params = array())
    {
        $query = array();
        if (isset($params['page'])) {
            $query['page'] = $params['page'];
        }
        if (isset($params['limit'])) {
            $query['limit'] = $params['limit'];
        }
        if (isset($params['status'])) {
            $query['status'] = $params['status'];
        }
        if (isset($params['domainId'])) {
            $query['domainId'] = $params['domainId'];
        }
        if (isset($params['from'])) {
            $query['from'] = $params['from'];
        }
        if (isset($params['to'])) {
            $query['to'] = $params['to'];
        }
        return $this->http->get('/logs', $query);
    }

    /**
     * Get a single mail-log entry by id.
     *
     * @param string $id
     * @return array
     */
    public function get($id)
    {
        return $this->http->get('/logs/' . rawurlencode($id));
    }
}

File: go/logs.go
package sentroy

import (
	"net/http"
	"net/url"
	"strconv"
)

// LogsService reads the mail send log.
type LogsService struct {
	h *httpClient
}

func newLogsService(h *httpClient) *LogsService {
	return &LogsService{h: h}
}

// List returns mail-log entries. Filter by status, domain, and ISO
// timestamp range (From / To, inclusive). Results are paginated
// server-side; pass Page and Limit to walk a large window.
func (s *LogsService) List(params *LogListParams) ([]MailLog, error) {
	q := map[string]string{}
	if params != nil {
		if params.Page > 0 {
			q["page"] = strconv.Itoa(params.Page)
		}
		if params.Limit > 0 {
			q["limit"] = strconv.Itoa(params.Limit)
		}
		if params.Status != "" {
			q["status"] = string(params.Status)
		}
		if params.DomainID != "" {
			q["domainId"] = params.DomainID
		}
		if params.From != "" {
			q["from"] = params.From
		}
		if params.To != "" {
			q["to"] = params.To
		}
	}
	return doRequest[[]MailLog](s.h, http.MethodGet, "/logs", q, nil)
}

// Get returns a single mail-log entry by id.
func (s *LogsService) Get(id string) (*MailLog, error) {
	l, err := doRequest[MailLog](s.h, http.MethodGet, "/logs/"+url.PathEscape(id), nil, nil)
	if err != nil {
		return nil, err
	}
	return &l, nil
}
